commons/pac-batch-commons/src/main/java/com/tmobile/pacman/commons/utils/NotificationPermissionUtils.java

```diff
@@ -14,19 +15,18 @@
 import java.util.Map;
 
 public class NotificationPermissionUtils {
-    private static final String NOTIFICATION_URL ="notification.lambda.function.url" ;
+    private static final String NOTIFICATION_URL = "notification.lambda.function.url";
     static String SUBJECT = "Insufficient permissions";
 
     static String opsEventName = "Permission denied for %s";
     private static PaladinAccessToken accessToken;
     private static final Logger LOGGER = LoggerFactory.getLogger(NotificationPermissionUtils.class);
-    public static void triggerNotificationsForPermissionDenied(List<PermissionVH> permissionVHList,String cloudType) {
+
+    public static void triggerNotificationsForPermissionDenied(List<PermissionVH> permissionVHList, String cloudType) {
         try {
             Gson gson = new Gson();
             List<NotificationBaseRequest> notificationDetailsList = new ArrayList<>();
-            for (PermissionVH permissionVH : permissionVHList) {
-                notificationDetailsList.add(getNotificationBaseRequest(permissionVH,cloudType));
-            }
+            notificationDetailsList.add(getNotificationBaseRequest(permissionVHList, cloudType));
```

Comment:
If the notification list is empty, we still want to create a record of the notification, sending an email is always optional.

Reply:
if the notification list is empty , what should the user be notified of ?

Reply:
There are two types of flow we have:
1. Is in-app notifications - default flow for all notifications. This flow delivers in-app notification + records the notification in OS
2. Notification delivery to external sources like email, jira, slack, etc

Reply:
These flows will be taken care in lambda functions. We will call the notification url only when there is any insufficient permission

Reply:
How? If the flow will not execute because of the if statement.